packages/bot-skeleton/src/scratch/blocks/Binary/Tools/Misc./console.js

```diff
@@ -0,0 +1,56 @@
+import { localize } from '@deriv/translations';
+import { emptyTextValidator } from '../../../../utils';
+
+Blockly.Blocks.console = {
+    init() {
+        this.jsonInit(this.definition());
+    },
+    definition() {
+        return {
+            message0: localize('Console %1 value: %2'),
+            args0: [
+                {
+                    type: 'field_dropdown',
+                    name: 'CONSOLE_TYPE',
+                    options: [
+                        [localize('Log'), 'log'],
+                        [localize('Warn'), 'warn'],
+                        [localize('Error'), 'error'],
+                        [localize('Table'), 'table'],
+                    ],
+                },
+                {
+                    type: 'input_value',
+                    name: 'MESSAGE',
+                    check: null,
+                },
+            ],
+            colour: Blockly.Colours.Special3.colour,
+            colourSecondary: Blockly.Colours.Special3.colourSecondary,
+            colourTertiary: Blockly.Colours.Special3.colourTertiary,
+            previousStatement: null,
+            nextStatement: null,
+            tooltip: localize('Displays a message in conosle developer tool'),
+            category: Blockly.Categories.Miscellaneous,
+        };
+    },
+    meta() {
+        return {
+            display_name: localize('Console'),
+            description: localize('This block console a message. You can see the message in console developer tool .'),
+        };
+    },
+    getRequiredValueInputs() {
+        return {
+            MESSAGE: emptyTextValidator,
+        };
+    },
+};
+
+Blockly.JavaScript.console = block => {
+    const consoleType = block.getFieldValue('CONSOLE_TYPE');
+    const message = Blockly.JavaScript.valueToCode(block, 'MESSAGE') || `"${localize('<empty message>')}"`;
```

Comment:
We don't have to supply a fallback value, the console will correctly log `undefined` if there's nothing defined.

Reply:
ah right